Adds Chemical Explosives

nullbear opened the pull request:
:cl: nullbear
rscadd: Added several craftable explosive chemical weapons.
rscadd: Made it possible to craft empty bomb assemblies using plasteel. Note: they will need a payload, and to have their wires set, before it can be detonated.
tweak: Large Grenades are 10% larger.
tweak: Grenades heat up their contents when they explode.
tweak: Non-grenade reagents are applied per turf affected, rather than per atom.
/:cl:

nullbear committed:
Adds Chemical Explosives

see changelog for details

Cheridan commented:
That's not how we do changelogs these days

nullbear commented:
what do you mean ? your wiki told me to do it like that.

Cheridan commented:
It's outdated then

nullbear commented:
what do i do then? The example changelog says to do this too.

ike709 commented:
see literally any other PR with a changelog

nullbear committed:
Builds and Updates changelog

nullbear commented:
Better?

Iamgoofball commented on code/game/objects/items/weapons/grenades/chem_grenade.dm:
```diff
@@ -272,6 +224,76 @@
+	affected_area = 3
+	ignition_temp = 500 // This is enough to expose a hotspot.
+
+/obj/item/weapon/grenade/chem_grenade/adv_release // Intended for weaker, but longer lasting effects. Could have some interesting uses.
```
HNNNNGH

:+1:

Bawhoppen commented:
don't bundle a bunch of mostly unrelated changes together, 
make a seperate PR for the stuff like buffing grenades, and the setting off bombs part
also don't make bombs be emaggable don't you dare

nullbear commented:
Well i made the bombs emaggable cause the minimum timer was changed from 60 to 120. So emaggable bombs just removes the nerf for antags.

As for the buffing grenades and stuff. That's all part of the chem_splash proc that they're all reliant on. Beforehand, the splash would do something like "units / 5-100,000+" where you'd end up with like 0.00001 units being applied to everything splashed. The 'buff' changes it to apply the units per turf, rather than per object, so it's much more consistent, and will typically have an actual effect instead of not doing anything. This was tested by filling a grenade with 600 units of acid in the bar. Using current grenades, nothing at all happens.

The rest of the unrelated stuff could be put to a different pr, sure. I figured they were related, though.

nullbear commented:
@gunhog how is this looking?

Iamgoofball commented:
ignore bawhoppen he's dumb

RemieRichards commented on code/game/machinery/syndicatebomb.dm:
```diff
@@ -292,6 +326,127 @@
+				B.loc = get_turf(src)
+
+		qdel(G)
+	return
```
trailing return, remove it please.

nullbear replied:
I don't know what that means.

GunHog replied:
You do not have to include the "return" command at the end of a proc, as it will do so automatically.

Supermichael777 commented:
do note that that timers is extremely inaccurate and 60 seconds typically translates into 2 minutes or more. also you could make the minimum time a var and have a sub-type with a higher number for crafted ones. emags do way to much already

nullbear commented:
Yeah, that's what i ended up going with, @Supermichael777

nullbear committed:
Removes nerf and emag

Sets minimum timer to initial timer, and resets to the initial time.

Removes 'trailing return'
